rfcs/text/0018_timeslider.md

```diff
@@ -0,0 +1,215 @@
+ 
+In addition, we expect the timeslider will be used a lot for "comparisons", ie. a user stepping back&forth between timeslices.
+
+For this reason, apps using a timeslider (such as Maps) should:
+- pre-fetch data when possible
+- cache data when possible
+
+For Maps specifically, when introducing timeslider, layers will therefore need to implement time-based data fetch based on _two_ pieces of state
+- the entire `timeRange` (aka. the global Kibana timerange)
+- the selected `timeslice` (aka. the timeslice chosen by the user in the timeslider)
+
+##### 2.3.3.1 Pre-fetching individual documents and masking of data
+
+ES-document layers (which display individual documents) can prefetch all documents within the entire `TimeRange`, when the total number of docs is belwo some threshold. In the context of Maps, this is the default index-search-size of the index.
+
+Maps can then just mask data on the map based on some filter-expression. The evaluation of this filter-expression is done by mapbox-gl is fast because it occurs on the GPU. There is immediate visual feedback to the user, without a network roundtrip to update the data.
+
+##### 2.3.3.2 Caching of aggregated searches
+
+
+Aggregated data can be cached on the client, so toggling between timeslices can avoid a round-trip data-fetch. 
+The main technique here is for layers to use the mvt to request data. Tiled-based data can be cached client-side
+
+We do not propose pre-fetching of aggregated data in this initial phase of the Maps timeslider effort. There is a couple reasons:
+- Based on the intended user-interactions for timeslider, where the end-user can flexible select a time-range, it would be really hard to determine how to determine which timeslices to aggregate up front.
+- Maps already strains the maximum bucket sizes we can retrieve from Elasticsearch. Cluster/grid-layers often push up to 10k or more buckets, and terms-aggregations for choropleth maps also is going up to 10k buckets. Prefetching this for timeslices (x10 timeslices) would easily exceed the default bucket limit.
+
+
+##### 2.3.3.3 Decouple data-fetch from UX-effort
+
+Apart from refactoring the data-fetch for layers to now use two pieces of time-based state, the implementation will decouple any data-fetch considerations from the actual timeslider-UX work.
+ 
+The idea is that dial in data-fetch optimizations can be dialed into Maps in a parallel work-thread, not necessarily dependent on any changes or additions to the UX. Any optimizations would not only affect timeslider users, but support all interaction patterns that require smooth data-updates (e.g. panning back&forth to two locations, toggling back&forth between two filters, ...)
+
+The main effort to support efficient data-fetch in a maps-context is move use `.mvt` as the default data format. This is a stack-wide effort in collaboration with the Elasticsearch team, which will add `.mvt` as a core response format to Elasticsearch.
```

Comment:
"The main effort to support efficient data-fetch in a maps-context is move use `.mvt` as the default data format" to "The main effort to support efficient data-fetch in a maps-context is to use `.mvt` as the default data format", replacing "move" with "to" from the sentence.